sourcegraph/sourcegraph
# Router V6 Migration: add CompatRouter and update Link component

This is a roll-back of a [rolled back original first PR](https://github.com/sourcegraph/sourcegraph/commit/d6f0c8779ac835bfb6f7771e860bd43c148e8c04) in the router v5 --> v6 migration sequence.

See the original for description and context: https://github.com/sourcegraph/sourcegraph/pull/36285

## What changed

`ReferencesPanel`, which crashed with the previous update because it was mistakenly wrapped in a `<CompatRouter>` too, is fixed.

For details, see slack threads:
- covering the [first bug report and the following roll-back](https://sourcegraph.slack.com/archives/C01C3NCGD40/p1655289377177509)
- explanation of [what's happened and why](https://sourcegraph.slack.com/archives/C01LTKUHRL3/p1655395878523209)

## Test plan
1. Pull the branch
2. Launch the app locally and make sure all main routes work
3. Check explicitly that the references panel works

## App preview:

- [Web](https://sg-web-og-router-migration-link.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ejpksgvryh.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
